components/brave_wallet/browser/simulation_request_helper.cc

```diff
@@ -43,13 +43,10 @@ base::Value::Dict GetMetadata(const mojom::OriginInfoPtr& origin_info) {
 namespace evm {
 
 std::optional<std::string> EncodeScanTransactionParams(
-    const mojom::TransactionInfo& tx_info) {
-  if (!tx_info.from_address) {
-    return std::nullopt;
-  }
-
+    const mojom::TransactionInfo& tx_info,
```

Comment:
Can we pass KeyringService to the constructor and use it to determine from address instead of passing address directly?

Reply:
If it appears there is no valid address for a given transaction that's a high level issue. So let caller handle that.